controller/metrics/metrics.go

```diff
@@ -381,7 +381,13 @@ func (c *appCollector) collectApps(ch chan<- prometheus.Metric, app *argoappv1.A
 		healthStatus = health.HealthStatusUnknown
 	}
 
-	addGauge(descAppInfo, 1, git.NormalizeGitURL(app.Spec.GetSource().RepoURL), app.Spec.Destination.Server, app.Spec.Destination.Namespace, string(syncStatus), string(healthStatus), operation)
+	autoSyncEnabled := false
+
+	if app.Spec.SyncPolicy != nil {
+		autoSyncEnabled = app.Spec.SyncPolicy.Automated.SelfHeal
```

Comment:
Both, `app.Spec.SyncPolicy` and `app.spec.SyncPolicy.Automated` are pointers and can be `nil`. Also, SelfHeal is not the correct value.
```suggestion
		autoSyncEnabled = app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil
```

Reply:
Updated pr to reflect recommended changes, thank you!